### PR TITLE
feat: add controlplane operations for vectorindex

### PR DESCRIPTION
**Pull request by virratanasangpunth**

### CODE REVIEW

**Review by PDXKimani — DISMISSED** [review]

File: proto/controlclient.proto [hunk 1]
```diff
@@ -15,6 +15,36 @@ service ScsControl {
...
+
+message _DeleteIndexResponse {
+}
+
+message _ListIndexesRequest {
+}
+
+message _ListIndexesResponse {
+  repeated _Index indexes = 1;
+  string next_token = 2;
```
Comment (nidhidhamnani) [inline proto/controlclient.proto:43]: What is `next_token` used for?

Comment (PDXKimani) [reply proto/controlclient.proto:43]: pagination

**Review by nidhidhamnani — DISMISSED** [review]

File: proto/controlclient.proto [hunk 2]
```diff
@@ -15,6 +15,36 @@ service ScsControl {
...
+}
+
+message _DeleteIndexRequest {
+  string index_name = 1;
+}
+
+message _DeleteIndexResponse {
+}
+
+message _ListIndexesRequest {
```
Comment (PDXKimani) [inline proto/controlclient.proto:38]: @virratanasangpunth though, if you're gonna do pagination, you need an optional token param here, right?

Comment (virratanasangpunth) [reply proto/controlclient.proto:38]: 🦅 👀

**Review by PDXKimani — APPROVED** [review]

**Review by mjindalmotiv8 — APPROVED** [review]

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Pull request merged**